browser/new_tab/brave_new_tab_service.cc

```diff
@@ -0,0 +1,36 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/new_tab/brave_new_tab_service.h"
+
+#include "chrome/common/webui_url_constants.h"
+#include "components/keyed_service/content/browser_context_dependency_manager.h"
+#include "content/public/browser/navigation_controller.h"
+#include "content/public/browser/web_contents.h"
+
+BraveNewTabService::BraveNewTabService(content::BrowserContext* browser_context)
+    : browser_context_(browser_context) {
+  PreloadNewTab();
```

Comment:
Does this result in consuming more memory by always holding +1 renderer process?
If so, do we ready to pay extra resource for this functionality?

BTW, we already have `Spare Renderer`, that is usually used for regular pages.

cc @iefremov

Reply:
Thanks for raising this issue! I actually tried the `Spare Renderer', but that doesn't help, unfortunately, because the goal is to avoid the annoying flicker, not to save some pre-warming time.
Anyway, this is just an experiment for early feedback from the advertising/new tabs/products teams, and we're still discussing whether we're ready for this on the product side. If we're ready for this behavior, I can improve some moments of this prototype and spend more time researching how to save resources.

Reply:
Thanks for the answer.
Let's discuss here: https://bravesoftware.slack.com/archives/C3T9S9WKD/p1658477270151989